Add frontmatter to the JSON result

mike-burns opened the pull request:
This adds all the frontmatter to the JSON result under the `meta` key.

Closes #11.

mike-burns committed:
Add frontmatter to the JSON result

This adds all the frontmatter to the JSON result under the `meta` key.

Closes #11.

mike-burns commented:
This was a pairing with Eric Moy and Andy at the NYC First-Time Open Source Contributor Workshop.

afeld commented on spec/integration_spec.rb:
```diff
@@ -94,6 +94,9 @@ def homepage_data
     expect(page['tags']).to eq(["Jekyll", "test page", "convenient"])
+    expect(page['meta']['layout']).to eq("page")
+    expect(page['meta']['permalink']).to eq("/about/")
+    expect(page['meta']['skip_index']).to eq(true)
```
Mind checking that the `title` and `tags` aren't included? Thinking it makes sense to not duplicate that information.

mike-burns replied:
Discussed IRL: duplicate the data (and include a test demonstrating this) to make life easier for machines.

afeld replied:
Talked a bit in person, and decided it was better to include them for consistency.

mike-burns committed:
Test that tags and title are in JSON metadata

After discussion IRL with Aidan, add tests for the `tags` and `title`
metadata in the JSON. This is to document that we intentionally
duplicate the tags and title.

We duplicate this because: as a human it makes sense to reduce
duplication, but as a machine I would expect to see all the frontmatter
in one place.

afeld commented on lib/jekyll_pages_api/page.rb:
```diff
@@ -59,11 +59,13 @@ def skip_index?
         tags: self.tags,
-        body: self.body_text
+        body: self.body_text,
+        meta: self.page.data
```
Mind adding this field to the example in the README?

https://github.com/18F/jekyll_pages_api#usage

mike-burns replied:
:+1:

mike-burns committed:
Document the new meta key

afeld commented:
Thanks for contributing everyone!
